cmd/kubeadm/app/apis/kubeadm/v1beta2/types.go

```diff
@@ -205,7 +205,7 @@ type NodeRegistrationOptions struct {
 	// Taints specifies the taints the Node API object should be registered with. If this field is unset, i.e. nil, in the `kubeadm init` process
 	// it will be defaulted to []v1.Taint{'node-role.kubernetes.io/master=""'}. If you don't want to taint your control-plane node, set this field to an
 	// empty slice, i.e. `taints: {}` in the YAML file. This field is solely used for Node registration.
-	Taints []v1.Taint `json:"taints,omitempty"`
+	Taints []v1.Taint `json:"taints"`
```

Comment:
why remove omitempty here?

Reply:
@luxas I think you may have missed the explanation in the PR description above. Here's an excerpt:

> It can be used in inappropriate places. In this case it's used for fields, that need to be always serialized. The only one such field at the moment is NodeRegistrationOptions.Taints. If the Taints field is nil, then it's defaulted to a slice containing a single control plane node taint. If it's an empty slice, no taints are applied, thus, the cluster behaves differently. With that in mind, a Go program, that uses v1beta1 with omitempty on the Taints field has no way to specify an explicit empty slice of taints, as this would get lost after marshalling to YAML.

Reply:
ah, ok, yeah I missed that :+1: